standards/ICRC-3/README.md

```diff
@@ -0,0 +1,288 @@
+
+## Blocks Verification
+
+The Ledger MUST certify the last block (tip) recorded. The Ledger MUST allow to download the certificate via the `icrc3_get_tip_certificate` endpoint. The certificate follows the [IC Specification for Certificates](https://internetcomputer.org/docs/current/references/ic-interface-spec#certification). The certificate is comprised of a tree containing the certified data and the signature. The tree MUST contain two labelled values (leafs):
+1. `last_block_index`: the index of the last block in the chain. The values must be expressed as [`leb128`](https://en.wikipedia.org/wiki/LEB128#Unsigned_LEB128)
+2. `last_block_hash`: the hash of the last block in the chain
+
+Clients SHOULD download the tip certificate first and then download the block backward starting from `last_block_index` and validate the blocks in the process.
+
+Validation of block `i` is done by checking the block hash against
+1. if `i + 1 < len(chain)` then the parent hash `phash` of the block `i+1`
+2. otherwise the `last_block_hash` in the tip certificate.
+
+## Generic Block Schema
+
+1. it MUST be a [`Value`](#value) of variant `Map`
+2. it MUST contain a field `tx: Map`
+    1. `tx` MUST contain a field `op: Text`, aka operation, which uniquely describes the type of the Block. `op` values `approve`, `burn`, `mint` and `xfer` are reserved for ICRC-1 and ICRC-2 Blocks
+2. it MUST contain a field `phash: Blob` which is the [hash](#value-hash) of its parent if it has a parent block
+
+## Interaction with other standards
+
+Each standard that adheres to `ICRC-3` MUST define the list of block schemas that it introduces. Each block schema MUST:
+
+1. extend the [Generic Block Schema](#generic-block-schema)
+2. specify the expected value of `tx.op`. This MUST be unique accross all the standards. `approve`, `burn`, `mint` and `xfer` are reserved for ICRC-1 and ICRC-2
+
+## [ICRC-1](../ICRC-1/README.md) and [ICRC-2](../ICRC-2/README.md) Block Schema
+
+ICRC-1 and ICRC-2 use the `tx` field to store input from the user and use the external block to store data set by the Ledger. For instance, the amount of a transaction is stored in the field `tx.amt` because it has been specified by the user, while the time when the block was added to the Ledger is stored in the field `ts` because it is set by the Ledger.
+
+A generic ICRC-1 or ICRC-2 Block:
+
+1. it MUST contain a field `ts: Nat` which is the timestamp of when the block was added to the Ledger
+2. if the `tx` field doesn't specify the fee then it MUST contain a field `fee: Nat` which specifies the fee payed to add this block to the Ledger
+3. its field `tx`
+    1. MUST contain a field `amt: Nat` that represents the amount
+    2. MUST contain the `fee: Nat` field if the user specifies the fee in the request. If the user does not specify the fee in the request, then this field is not set and the top-level `fee` is set.
```

Comment:
```suggestion
    2. MUST contain the `fee: Nat` field for operations that require a fee if the user specifies the fee in the request. If the user does not specify the fee in the request, then this field is not set and the top-level `fee` is set.
```

Reply:
All ICRC-1 and ICRC-2 require a fee.

Reply:
even mint and burn?

Reply:
There is no fee field in the burn or mint examples.

Reply:
Best to specify which operations require a fee field in icrc1/2 if mint and burn does not.